src/p2lab/genetic/matching.py

```diff
@@ -0,0 +1,44 @@
+import networkx as nx
+import numpy as np
+
+def dense(teams : list[object]) -> np.ndarray:
+	r"""
+	Generates a fully connected set of edges. Each team is 
+	connected to every other one.
+
+	Inputs:
+		:teams: set of teams.
+
+	Outputs:
+	
+		np.ndarray shape (N_edges, 2)
+		
+	"""
+	g = nx.complete_graph(len(teams))
+
+	# Select only the node index pairs, exclude the feature dict.
+	return np.stack( g.edges(data=False) )
+
+def mst(teams : list[object]) -> np.ndarray:
+	r"""
+	Returns a minimum spanning tree graph generated from the
+	fully connected graph over the teams.
+
+	Outputs:
+
+		np.ndarray shape (N_edges, 2)
+
+	"""
+	g = nx.complete_graph(len(teams))
+	mst = nx.algorithms.tree.minimum_spanning_edges(g, data=False)
+
+	return list(mst)
+
+def main():
+	
+	# Test
+	print(mst([0] * 5))
+	print(dense([0] * 5))
+
+if __name__ == "__main__":
+	main()
```

Comment:
Are you still using these tests or were these used for writing the function? If still in use, I think we're using pytest in this repo. If so, move to a test_matching.py file in root/tests/

Reply:
Okay - I leave it up to you whether or not to implement this now or to leave it as a potential addition later. If later, this looks ready to merge to me!

One thought is it could be parameterised as a value between 0 and 1, where 0 means we have M edges, 1 means we have N, 0.5 T = (N+M)/2 edges, etc